src/rest.cpp

```diff
@@ -242,11 +243,7 @@ static bool rest_block(HTTPRequest* req,
     }
 
     case RF_JSON: {
-        UniValue objBlock;
-        {
-            LOCK(cs_main);
-            objBlock = blockToJSON(block, pblockindex, showTxDetails);
-        }
+        UniValue objBlock = std::move(blockToJSON(block, tip, pblockindex, showTxDetails));
```

Comment:
Does return-value optimization not capture this?

Reply:
I think you are correct.